execution_engine2.spec

```diff
@@ -67,72 +67,58 @@
         /*  Running long running methods through Docker images of services from Registry  */
         /*================================================================================*/
         /* A workspace object reference of the form X/Y or X/Y/Z, where
-           X is the workspace name or id,
-           Y is the object name or id,
-           Z is the version, which is optional.
+           X is the workspace id,
+           Y is the object id,
+           Z is the version.
          */
         typedef string wsref;
 
-        /*
-            time - the time the call was started;
-            method - service defined in standard JSON RPC way, typically it's
-                module name from spec-file followed by '.' and name of funcdef
-                from spec-file corresponding to running method (e.g.
-                'KBaseTrees.construct_species_tree' from trees service);
-            job_id - job id if method is asynchronous (optional field).
-        */
-        typedef structure {
-            timestamp time;
-            string method;
-            job_id job_id;
-        } MethodCall;
-
-        /*
-            call_stack - upstream calls details including nested service calls and
-                parent jobs where calls are listed in order from outer to inner.
+        /* Narrative metadata for a job. All fields are optional.
+            run_id - the Narrative-assigned ID of the job run. 1:1 with a job ID.
+            token_id - the ID of the token used to run the method.
+            tag - the release tag, e.g. dev/beta/release.
+            cell_id - the ID of the narrative cell from which the job was run.
         */
         typedef structure {
-            list<MethodCall> call_stack;
             string run_id;
-        } RpcContext;
+            string token_id;
+            string tag;
+            string cell_id;
+        } Meta;
 
         /*
-            method - service defined in standard JSON RPC way, typically it's
-                module name from spec-file followed by '.' and name of funcdef
-                from spec-file corresponding to running method (e.g.
-                'KBaseTrees.construct_species_tree' from trees service);
-            params - the parameters of the method that performed this call;
+            method - the SDK method to run in module.method format, e.g.
+                'KBaseTrees.construct_species_tree'
+            app_id - the id of the Narrative application (UI) running this job (e.g.
```

Comment:
app id is actually required, so moved up from the optional section

Reply:
App id is not required, as a UI is not required to run a job.

Reply:
I always thought it was weird that it was required, but trying to run a job without it fails because the DB model specifies it as required:
https://github.com/kbase/execution_engine2/blob/develop/lib/execution_engine2/db/models/models.py#L167